ts/Series/Lollipop/LollipopSeries.ts

```diff
@@ -0,0 +1,259 @@
+    }
+} = BaseSeries;
+import U from '../../Core/Utilities.js';
+const {
+    pick,
+    merge,
+    extend
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * lollipop series type
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.lollipop
+ *
+ * @augments Highcharts.Series
+ *
+ */
+class LollipopSeries extends DumbbellSeries {
+
+    /* *
+     *
+     *  Static properties
+     *
+     * */
+
+    /**
+     * The lollipop series is a carteseian series with a line anchored from
+     * the x axis and a dot at the end to mark the value.
+     * Requires `highcharts-more.js`, `modules/dumbbell.js` and
+     * `modules/lollipop.js`.
+     *
+     * @sample {highcharts} highcharts/demo/lollipop/
+     *         Lollipop chart
+     * @sample {highcharts} highcharts/series-dumbbell/styled-mode-dumbbell/
+     *         Styled mode
+     *
+     * @extends      plotOptions.dumbbell
+     * @product      highcharts highstock
+     * @excluding    fillColor, fillOpacity, lineWidth, stack, stacking,
+     *               lowColor, stickyTracking, trackByArea
+     * @since 8.0.0
+     * @optionparent plotOptions.lollipop
+     */
+    public static defaultOptions: LollipopSeriesOptions = merge(
+        DumbbellSeries.defaultOptions,
+        {
+            /** @ignore-option */
+            lowColor: void 0,
+            /** @ignore-option */
+            threshold: 0,
+            /** @ignore-option */
+            connectorWidth: 1,
+            /** @ignore-option */
+            groupPadding: 0.2,
+            /** @ignore-option */
+            pointPadding: 0.1,
+            /** @ignore-option */
+            states: {
+                hover: {
+                    /** @ignore-option */
+                    lineWidthPlus: 0,
+                    /** @ignore-option */
+                    connectorWidthPlus: 1,
+                    /** @ignore-option */
+                    halo: false
+                }
+            },
+            tooltip: {
+                pointFormat: '<span style="color:{series.color}">●</span> {series.name}: <b>{point.y}</b><br/>'
+            }
+        } as LollipopSeriesOptions);
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<LollipopPoint> = void 0 as any;
+    public options: LollipopSeriesOptions = void 0 as any;
+    public points: Array<LollipopPoint> = void 0 as any;
+
+}
+
+/* *
+ *
+ *  Prototype properties
+ *
+ * */
+
+interface LollipopSeries {
+    pointClass: typeof LollipopPoint;
+    pointArrayMap: Array<string>;
+    pointValKey: string;
+    translatePoint: typeof areaProto['translate'];
+    drawPoint: typeof areaProto['drawPoints'];
+    drawDataLabels: typeof colProto['drawDataLabels'];
+    setShapeArgs: typeof colProto['translate'];
+}
+
+extend(LollipopSeries.prototype, {
+    pointArrayMap: ['y'],
+    pointValKey: 'y',
+    toYData: function (point: LollipopPoint): Array<number> {
+        return [pick(point.y, point.low)];
+    },
```

Comment:
What should I do about the return type? The inherited type is a `[number, number]` tuple. Should I just `any`-cast this or change the type in AreaRangeSeries?

Reply:
It is better for inheritance to avoid tuples, otherwise you have to specify all the different tuple sizes down the road, even so in its core it is just an array.